tools/asg-hevc/src/block_structures.cpp

```diff
@@ -193,12 +193,12 @@ PatchBlock::PatchBlock(const BaseBlock& dstBlock, const PatchBlock& srcBlock) :
 
     for (mfxU32 i = 0; i < m_BHeight; i++)
     {
-        memcpy(m_YPlane + i * m_BWidth, srcBlock.m_YPlane + (offsetY + i) * srcBlock.m_BWidth + offsetX, m_BWidth);
+        std::copy(srcBlock.m_YPlane + (offsetY + i) * srcBlock.m_BWidth + offsetX, srcBlock.m_YPlane + (offsetY + i) * srcBlock.m_BWidth + offsetX + m_BWidth, m_YPlane + i * m_BWidth);
```

Comment:
this looks to complicated and very error-prone...
We need a redesign of this part, maybe in next commits